src/applications/personalization/profile360/util/index.js

```diff
@@ -8,3 +23,98 @@ export async function getData(apiRoute, options) {
     return { error };
   }
 }
+
+function hasErrorMessageKey(errors, errorKey) {
+  return errors.some(err =>
+    err.meta.messages.some(message => message.key === errorKey),
+  );
+}
+
+function hasErrorMessageText(errors, errorText) {
+  return errors.some(err =>
+    err.meta.messages.some(message =>
+      message.text.toLowerCase().includes(errorText.toLowerCase()),
+    ),
+  );
+}
+
+export function hasFlaggedForFraudError(errors) {
+  return (
+    hasErrorMessageKey(errors, ACCOUNT_FLAGGED_FOR_FRAUD_KEY) ||
+    hasErrorMessageKey(errors, PAYMENT_RESTRICTIONS_PRESENT_KEY) ||
+    hasErrorMessageKey(errors, ROUTING_NUMBER_FLAGGED_FOR_FRAUD_KEY)
+  );
+}
+
+export function hasInvalidRoutingNumberError(errors) {
+  let result = false;
+  if (hasErrorMessageKey(errors, INVALID_ROUTING_NUMBER_KEY)) {
+    result = true;
+  }
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    hasErrorMessageText(errors, 'Invalid Routing Number')
+  ) {
+    result = true;
+  }
+  return result;
+}
+
+export function hasInvalidAddressError(errors) {
+  let result = false;
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    hasErrorMessageText(errors, 'address update')
+  ) {
+    result = true;
+  }
+  return result;
+}
+
+export function hasInvalidHomePhoneNumberError(errors) {
+  let result = false;
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    (hasErrorMessageText(errors, 'night phone number') ||
+      hasErrorMessageText(errors, 'night area number'))
+  ) {
+    result = true;
+  }
+  return result;
+}
+
+export function hasInvalidWorkPhoneNumberError(errors) {
+  let result = false;
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    (hasErrorMessageText(errors, 'day phone number') ||
+      hasErrorMessageText(errors, 'day area number'))
+  ) {
+    result = true;
+  }
+  return result;
+}
+
+// Helper that creates and returns an object to pass to the recordEvent()
+// function when an errors occurs while trying to save/update a user's direct
+// deposit payment information. The value of the `error-key` prop will change
+// depending on the content of the `errors` array.
+export function createEventDataObjectWithErrors(errors) {
+  const key = 'error-key';
+  const eventDataObject = {
+    event: 'profile-edit-failure',
+    'profile-action': 'save-failure',
+    'profile-section': 'direct-deposit-information',
+    [key]: GA_ERROR_KEY_DEFAULT,
+  };
```

Comment:
Is this exclusively used just for direct deposit, or are there plans to expand it? Asking bc `createEventDataObjectWithErrors` sounds generic versus something like `createDirectDepositErrorAnalytic` or something.

Also - apologies for the inconsistent naming of "payment information" vs "direct deposit." The former was the original name I was on eBenefits or in our mockups, so I thought that's what it was going to be.

Reply:
ah, good call. we're only using this for the Direct Deposit errors. I'll change the name of the helper.

Maybe I'll also change the references to PaymentInformation in the code when I do some refactoring.